examples/bank/bank.go

```diff
@@ -0,0 +1,214 @@
+
+import (
+	"fmt"
+	"math/rand"
+	"strconv"
+	"time"
+
+	"github.com/cockroachdb/cockroach/client"
+	"github.com/cockroachdb/cockroach/proto"
+	"github.com/cockroachdb/cockroach/security"
+	"github.com/cockroachdb/cockroach/security/securitytest"
+	"github.com/cockroachdb/cockroach/server"
+	"github.com/cockroachdb/cockroach/storage"
+	"github.com/cockroachdb/cockroach/testutils"
+	"github.com/cockroachdb/cockroach/util/log"
+)
+
+// makes an id string from an id int.
+func makeAccountID(id int) []byte {
+	return []byte(fmt.Sprintf("%09d", id))
+}
+
+// Bank stores all the bank related state.
+type Bank struct {
+	servingAddr string
+	numAccounts int
+	// This is not protected, but we really don't care.
+	approxNumTransactions int
+}
+
+// moveMoney() moves an amount between two accounts if the amount
+// is available in the from account. Returns true on success.
+func (bank *Bank) moveMoney(kvClient *client.KV, from, to []byte, amount int64) bool {
+	txnOpts := &client.TransactionOptions{Name: fmt.Sprintf("Transferring %d", amount)}
+	err := kvClient.RunTransaction(txnOpts, func(txn *client.Txn) error {
+		call := client.Get(proto.Key(from))
+		gr := call.Reply.(*proto.GetResponse)
+		if err := txn.Run(call); err != nil {
+			return err
+		}
+		// Read from value.
+		var fromValue int64
+		if gr.Value != nil && gr.Value.Bytes != nil {
+			readValue, err := strconv.ParseInt(string(gr.Value.Bytes), 10, 64)
+			if err != nil {
+				return err
+			}
+			fromValue = readValue
+		}
+
+		// Ensure there is enough cash.
+		if fromValue < amount {
+			return nil
+		}
+
+		// Read to value.
+		call = client.Get(proto.Key(to))
+		gr = call.Reply.(*proto.GetResponse)
+		if err := txn.Run(call); err != nil {
+			return err
+		}
+
+		var toValue int64
+		if gr.Value != nil && gr.Value.Bytes != nil {
+			readValue, err := strconv.ParseInt(string(gr.Value.Bytes), 10, 64)
+			if err != nil {
+				return err
+			}
+			toValue = readValue
+		}
+		// Update both accounts.
+		txn.Prepare(client.Put(proto.Key(from), []byte(fmt.Sprintf("%d", fromValue-amount))))
+		txn.Prepare(client.Put(proto.Key(to), []byte(fmt.Sprintf("%d", toValue+amount))))
+		return nil
+	})
+	if err != nil {
+		log.Fatal(err)
+	}
+
+	bank.approxNumTransactions++
+	return true
+}
+
+// Read the balances in all the accounts and return them.
+func (bank *Bank) readAllAccounts() []int64 {
+	balances := make([]int64, bank.numAccounts)
+	// Key Value Client initialization.
+	sender, err := client.NewHTTPSender(bank.servingAddr, testutils.NewTestBaseContext())
+	if err != nil {
+		log.Fatal(err)
+	}
+	kvClient := client.NewKV(nil, sender)
+	kvClient.User = storage.UserRoot
+	txnOpts := &client.TransactionOptions{Name: "Reading all balances"}
+	err = kvClient.RunTransaction(txnOpts, func(txn *client.Txn) error {
+		for i := 0; i < bank.numAccounts; i++ {
+			call := client.Get(proto.Key(makeAccountID(i)))
+			gr := call.Reply.(*proto.GetResponse)
+			if err := txn.Run(call); err != nil {
```

Comment:
Like elsewhere, you should perform all of the gets in parallel. Otherwise this is `O(num-accounts)` round-trips to the server.